src/main/java/com/google/sps/travelbud/Event.java

```diff
@@ -79,6 +79,19 @@ public static Event getEvent(DatastoreService datastore, long eventId) {
     }
   }
 
+  public static List<Event> filter(DatastoreService datastore, String name) {
+    List<Event> filteredEvents = new ArrayList<>();
+    List<Event> events = Event.getAll(datastore);
+    for (Event E : events) {
+      String nameOfE = E.getName().toLowerCase();
+      String substr = nameOfE.substring(0, name.length());
+      if (nameOfE.length() >= name.length() && substr.equals(name.toLowerCase())) {
```

Comment:
As an FYI, you can more easily do this like this:
```java
if (E.getName().toLowerCase().contains(name.toLowerCase())) {
```

also `E` shouldn't be a variable name as it's lower cased.